crates/ruma-client-api/src/r0/sync/sync_events.rs

```diff
@@ -327,6 +332,31 @@ impl Default for JoinedRoom {
     }
 }
 
+/// Updates to knocked rooms.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct KnockedRoom {
```

Comment:
Please make the two new structs (conditionally) non-exhaustive like almost all other structs in ruma.